src/pnm/encoder.rs

```diff
@@ -275,42 +276,46 @@ impl<'a> CheckedImageBuffer<'a> {
         width: u32,
         height: u32,
         color: ColorType,
-    ) -> io::Result<CheckedImageBuffer<'a>> {
+    ) -> ImageResult<CheckedImageBuffer<'a>> {
         let components = channel_count(color) as usize;
         let uwidth = width as usize;
         let uheight = height as usize;
         match Some(components)
             .and_then(|v| v.checked_mul(uwidth))
             .and_then(|v| v.checked_mul(uheight))
         {
-            None => Err(io::Error::new(
+            None => Err(ImageError::IoError(io::Error::new(
```

Comment:
There is `DimensionError` but that looses some information on the error. Probably discuss in #928 for some details?